scripts/check_compliance.py

```diff
@@ -542,61 +546,69 @@ def set_status(repo, sha):
     github_token = os.environ['GH_TOKEN']
     github_conn = Github(github_token)
 
-    repo = github_conn.get_repo(repo)
+    repo = github_conn.get_repo(repo_name)
     commit = repo.get_commit(sha)
     for testcase in ComplianceTest.__subclasses__():
         test = testcase(None, "")
-        print("Creating status for %s" % (test._name))
+        info("Creating status for %s" % (test._name))
         commit.create_status('pending',
                              '%s' % test._doc,
                              'Checks in progress',
                              '{}'.format(test._name))
 
 
-def report_to_github(repo, pull_request, sha, suite, docs):
+def report_to_github(repo_name, pull_request, sha, suite, docs, dryrun):
     """
     Report test results to Github
 
-    :param repo: repo name
+    :param repo_name: repo name
     :param pull_request:  pull request number
     :param sha:  pull request SHA
     :param suite:  Test suite
     :param docs:  documentation of statuses
     :return: nothing
     """
 
-    if 'GH_TOKEN' not in os.environ:
-        return
-
     username = os.environ.get('GH_USERNAME', 'zephyrbot')
+    github_token = os.environ.get('GH_TOKEN', None)
+    repo = None
+    gh_pr = None
+    commit = None
 
-    github_token = os.environ['GH_TOKEN']
-    github_conn = Github(github_token)
 
-    repo = github_conn.get_repo(repo)
-    gh_pr = repo.get_pull(pull_request)
-    commit = repo.get_commit(sha)
+    if not dryrun:
+        if github_token:
+            github_conn = Github(github_token)
+        else:
+            error("No github token")
```

Comment:
not sure what you mean.

Reply:
I mean if the script can't do it "for real" because of some token/authentication/other issue, why not behave exactly like the user had asked for a --dry-run as a fallback and an attempt to do something productive anyway?
The only risk I can foresee is the user thinking it did happen for real when it did not.